packages/editor-ui/src/components/__tests__/PersonalizationModal.spec.ts

```diff
@@ -61,4 +87,50 @@ describe('PersonalizationModal.vue', () => {
 			});
 		}
 	});
+
+	it('should display self serve trial option when company size is larger than 500', async () => {
+		const wrapper = renderComponent();
+
+		await retry(() =>
+			expect(wrapper.container.querySelector('.modal-content')).toBeInTheDocument(),
+		);
+
+		const select = wrapper.container.querySelectorAll('.n8n-select')[3]!;
+		await fireEvent.click(select);
+
+		const item = select.querySelectorAll('.el-select-dropdown__item')[3];
+		await fireEvent.click(item);
+
+		await retry(() => {
+			expect(wrapper.container.querySelector('.card')).not.toBeNull();
+		});
+	});
+
+	it('should display send telemetry when requesting enterprise trial', async () => {
+		const usersStore = useUsersStore(pinia);
+		vi.spyOn(usersStore, 'submitPersonalizationSurvey').mockResolvedValue();
+
+		const usageStore = useUsageStore(pinia);
+		const spyLicenseTrial = vi.spyOn(usageStore, 'requestEnterpriseLicenseTrial');
+
+		const wrapper = renderComponent();
+
+		await retry(() =>
+			expect(wrapper.container.querySelector('.modal-content')).toBeInTheDocument(),
+		);
+
+		const select = wrapper.container.querySelectorAll('.n8n-select')[3]!;
+		await fireEvent.click(select);
+
+		const item = select.querySelectorAll('.el-select-dropdown__item')[3];
+		await fireEvent.click(item);
+
+		const agreeCheckbox = wrapper.container.querySelector('.n8n-checkbox')!;
+		await fireEvent.click(agreeCheckbox);
+
+		const submitButton = wrapper.container.querySelector('button')!;
```

Comment:
I know and hate our select component is not accessible but for other elements we usually use the role selector
And also prefer using `user-event` library for dispatching events
In case of button:
```ts
import userEvent from '@testing-library/user-event';

const { getByRole } = renderComponent();
const submitButton = getByRole('button')!;

await userEvent.click(submitButton);
```

Reply:
Ah thanks for clarifying, changed and updated. Could you please approve again @cstuncsik ?